OpenEdXMobile/src/main/java/org/edx/mobile/view/BaseCourseUnitVideoFragment.java

```diff
@@ -181,14 +181,16 @@ public void setUserVisibleHint(boolean isVisibleToUser) {
 
     @Override
     public void downloadTranscript() {
-        TranscriptModel transcript = getTranscriptModel();
-        String transcriptUrl = LocaleUtils.getTranscriptURL(getActivity(), transcript);
-        transcriptManager.downloadTranscriptsForVideo(transcriptUrl, (TimedTextObject transcriptTimedTextObject) -> {
-            subtitlesObj = transcriptTimedTextObject;
-            if (!getActivity().isDestroyed()) {
-                initTranscripts();
-            }
-        });
+        if (getActivity() != null) {
+            TranscriptModel transcript = getTranscriptModel();
+            String transcriptUrl = LocaleUtils.getTranscriptURL(getActivity(), transcript);
+            transcriptManager.downloadTranscriptsForVideo(transcriptUrl, (TimedTextObject transcriptTimedTextObject) -> {
+                subtitlesObj = transcriptTimedTextObject;
+                if (!getActivity().isDestroyed()) {
+                    initTranscripts();
+                }
+            });
+        }
     }
```

Comment:
After this logic, we are still able to produce intentionally by following steps,
- Comment out the statement at [Line# 238](https://github.com/edx/edx-app-android/blob/omer/LEARNER-7721/OpenEdXMobile/src/main/java/org/edx/mobile/view/BaseCourseUnitVideoFragment.java#L238)
- Add getActivity().finish() after [Line # 186](https://github.com/edx/edx-app-android/pull/1404/files#diff-ad6f9e21b8cadc6575f66d73b585b863R186)

that can be fix by usin the following block of code:
```java
    @Override
    public void downloadTranscript() {
        Activity activity = getActivity();
        if (activity != null) {
            TranscriptModel transcript = getTranscriptModel();
            String transcriptUrl = LocaleUtils.getTranscriptURL(activity, transcript);
            transcriptManager.downloadTranscriptsForVideo(transcriptUrl, (TimedTextObject transcriptTimedTextObject) -> {
                subtitlesObj = transcriptTimedTextObject;
                if (!activity.isDestroyed()) {
                    initTranscripts();
                }
            });
        }
    }
```